internal/status/ing_status.go

```diff
@@ -261,15 +260,25 @@ func isAnnotationsUpdateRequired(ingAnnotations map[string]string, newVSAnnotati
 			return true
 		}
 	}
+	if !isDelete {
+		oldTenant, ok := ingAnnotations[lib.TenantAnnotation]
+		if !ok {
+			return true
+		}
+		if oldTenant != newTenant {
+			return true
+		}
+	}
 	return false
 }
 
-func getAnnotationsPayload(vsAnnotations map[string]string, existingAnnotations map[string]string) ([]byte, error) {
-	var vsAnnotationVal, ctrlAnnotationVal *string
+func getAnnotationsPayload(vsAnnotations map[string]string, tenant string) ([]byte, error) {
+	var vsAnnotationVal, ctrlAnnotationVal, tenantVal *string
 	ctrlAnnotationValStr := avicache.GetControllerClusterUUID()
 	if len(vsAnnotations) == 0 {
 		vsAnnotationVal = nil
 		ctrlAnnotationVal = nil
+		tenantVal = nil
```

Comment:
I think we can skip this `if` section as these vars would already be nil. We can only init the vars if len(vsAnnotations) > 0
Also, if vsAnnotation is empty we canjust return from there.

Reply:
updated